file_loader.py

```diff
@@ -0,0 +1,328 @@
+
+Input file structure:
+
+    └── source_data
+        └── dir_20231010
+            ├── MRc.1.3.12.2.543543
+            ├── ...
+            └── SRe. 1.3.12.2.5432233
+
+Output file structure:
+
+    ├── bids
+    │   └── sub-001
+    │       └── ses-01
+    │           ├── anat
+    │           │   ├── sub-001_ses-01_T2w.json
+    │           │   └── sub-001_ses-01_T2w.nii.gz
+    │           └── dwi
+    │               ├── sub-001_ses-01_dwi.bval
+    │               ├── sub-001_ses-01_dwi.bvec
+    │               ├── sub-001_ses-01_dwi.json
+    │               └── sub-001_ses-01_dwi.nii.gz
+    └── source_data
+        └── dir_20231010
+            ├── MRc.1.3.12.2.543543
+            ├── ...
+            └── SRe. 1.3.12.2.5432233
+
+Author: Jan Valosek and Claude 3.5 Sonnet
+"""
+
+# TODO: write a new entry to the participant.tsv file based on the provided information (add sex and age args?)
+# TODO: use logging instead of print statements
+
+
+import os
+import shutil
+import argparse
+import pandas as pd
+import nibabel as nib
+
+
+def get_parser():
+    """
+    Parse command-line arguments.
+
+    Returns:
+    argparse.Namespace: Parsed arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="Convert DICOM to NIfTI and identify images for the further analysis."
+    )
+    parser.add_argument(
+        "-dicom-folder",
+        help="Path to the folder containing DICOM images. "
+             "Example: ~/sci-balgrist-study/sourcedata/dir_20230711",
+        required=True
+    )
+    parser.add_argument(
+        "-bids-folder",
+        help="Path to the BIDS folder where the converted NIfTI images will be stored. "
+             "Example: ~/sci-balgrist-study/bids",
+        required=True
+    )
+    parser.add_argument(
+        "-participant",
+        help="Participant ID. Example: sub-001",
+        required=True
+    )
+    parser.add_argument(
+        "-session",
+        help="Session ID. Example: ses-01",
+        required=True
+    )
+    parser.add_argument(
+        "-contrasts",
+        help="MRI contrasts to use. Example: T2w dwi",
+        nargs='+',
+        default=["T2w", "dwi"],
+        type=list,
+        required=False
+    )
+    parser.add_argument(
+        "-debug",
+        help="If used, the temporary folder with NIfTI images will NOT be removed.",
+        action="store_true",
+        default=False,
+        required=False
+    )
+
+    return parser.parse_args()
+
+
+def get_image_info(file_path):
+    """
+    Get the dimensions and pixel size of the image at the given file path.
+
+    :param file_path: Path to the image file
+    """
+    img = nib.load(file_path)
+    zooms = img.header.get_zooms()
+
+    dimensions = f"{img.shape[0]} x {img.shape[1]} x {img.shape[2]}"
+    pixel_size = f"{zooms[0]:.2f} mm x {zooms[1]:.2f} mm x {zooms[2]:.2f} mm"
+
+    return dimensions, pixel_size
+
+
+def run_dcm2niix(dicom_folder, temp_folder):
```

Comment:
You should maybe check if dcm2niix is installed? and exit if it is not?

Reply:
Agree! I was initially thinking about running this Python script as part of the whole pipeline (i.e., calling the script from another script). The first step of the pipeline would check for the dependencies:

```bash
# Check if the necessary dependencies are installed
check_dependencies()
{
    # List of commands to check
    commands=("dcm2niix" "sct_check_dependencies")

    # Loop through each command and check if it is installed and callable
    for cmd in "${commands[@]}"; do
        if ! command -v $cmd &> /dev/null; then
            echo "$cmd could not be found. Please install it before running this script. Exiting..."
            exit 1
        fi
    done
}
```

We'll see how we structure the whole pipeline. I'm leaving this comment open so I don't forget.